src/Workspaces/Core/Portable/Notification/GlobalOperationNotificationService.cs

```diff
@@ -52,38 +52,34 @@ public override GlobalOperationRegistration Start(string operation)
 
         protected virtual Task RaiseGlobalOperationStarted()
         {
-            using (_listener.BeginAsyncOperation("GlobalOperationStarted"))
+            var ev = _eventMap.GetEventHandlers<EventHandler>(GlobalOperationStartedEventName);
+            if (ev.HasHandlers)
             {
-                var ev = _eventMap.GetEventHandlers<EventHandler>(GlobalOperationStartedEventName);
-                if (ev.HasHandlers)
+                var asyncToken = _listener.BeginAsyncOperation("GlobalOperationStarted");
+                return _eventQueue.ScheduleTask(() =>
                 {
-                    return _eventQueue.ScheduleTask(() =>
-                    {
-                        ev.RaiseEvent(handler => handler(this, EventArgs.Empty));
-                    });
-                }
-
-                return SpecializedTasks.EmptyTask;
+                    ev.RaiseEvent(handler => handler(this, EventArgs.Empty));
+                }).CompletesAsyncOperation(asyncToken);
```

Comment:
:memo: This is definitely a better use of `IAsyncToken`, but note that this is a change to production code so it's not quite test-only.

Reply:
true. but IAsyncToken is no-op in production.